includes/Core/Modules/Modules.php

```diff
@@ -252,17 +252,13 @@ function( $token_response ) {
 					return;
 				}
 
-				if ( ! $this->is_module_active( Analytics::MODULE_SLUG ) ) {
-					$this->activate_module( Analytics::MODULE_SLUG );
-				}
-
-				try {
-					$analytics = $this->get_module( Analytics::MODULE_SLUG );
-					$analytics->handle_token_response_data( $token_response );
-				} catch ( Exception $e ) {
+				// Do nothing if the Analytics module is already activated.
+				if ( $this->is_module_active( Analytics::MODULE_SLUG ) ) {
 					return;
 				}
 
+				$this->activate_module( Analytics::MODULE_SLUG );
+
 				$extra_scopes = $this->user_options->get( OAuth_Client::OPTION_ADDITIONAL_AUTH_SCOPES );
 				if ( is_array( $extra_scopes ) && in_array( Analytics::READONLY_SCOPE, $extra_scopes, true ) ) {
 					unset( $extra_scopes[ array_search( Analytics::READONLY_SCOPE, $extra_scopes, true ) ] );
```

Comment:
Nit-pick: We could _only_ call `array_search` instead of both `array_search` and `in_array` here, and the condition could then just check if the found index is `> -1`. But nothing blocking.